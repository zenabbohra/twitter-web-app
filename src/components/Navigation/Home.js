import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import Link from '@material-ui/core/Link';
import HomeIcon from '@material-ui/icons/Home';

const useStyles = makeStyles(theme => ({
  link: {
    display: 'flex',

  },
  icon: {
    marginRight: theme.spacing(0.5),
    width: '1.1rem',
    height: '1.1rem',
  },
}));

export default function Home() {
  const classes = useStyles();

  return (
    <Link href="/" className={classes.link} color="primary" variant="body2">
      <HomeIcon className={classes.icon}/>
      Home
    </Link>
  );
}