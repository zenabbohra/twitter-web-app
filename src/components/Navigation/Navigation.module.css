.link {
    line-height: 1.9 !important;
    display: flex;
}

.icon {
    margin-right: 4px;
    width: 1.1rem;
    height: 1.1rem;
}