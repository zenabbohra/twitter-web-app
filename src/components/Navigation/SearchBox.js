import React from 'react';
import InputBase from '@material-ui/core/InputBase';
import {fade, makeStyles} from '@material-ui/core/styles';
import SearchIcon from '@material-ui/icons/Search';
import Box from "@material-ui/core/Box";

const useStyles = makeStyles(theme => ({
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25),
    },
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      width: 'auto',
    },
  },
  searchIcon: {
    width: theme.spacing(4),
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputRoot: {
    color: 'inherit',
  },
  inputInput: {
    padding: theme.spacing(1, 1, 1, 4),
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      width: 120,
      '&:focus': {
        width: 200,
      },
    },
  },
}));

export default function SearchBox() {
  const classes = useStyles();

  return (
    <Box flexGrow={1} border={1} borderColor="primary.main" borderRadius={64} padding={1}>
      <div className={classes.search}>
        <div className={classes.searchIcon}>
          <SearchIcon/>
        </div>
        <InputBase
          placeholder="Search"
          classes={{
            root: classes.inputRoot,
            input: classes.inputInput,
          }}
          inputProps={{'aria-label': 'search'}}
        />
      </div>
    </Box>
  );
}