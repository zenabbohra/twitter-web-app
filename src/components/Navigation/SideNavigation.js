import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import InboxIcon from '@material-ui/icons/Inbox';
import NotificationsIcon from '@material-ui/icons/Notifications';
import AccountBoxIcon from '@material-ui/icons/AccountBox';
import ExploreIcon from '@material-ui/icons/Explore';
import Divider from "@material-ui/core/Divider";


const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper,
    paddingTop: 0
  },
  nested: {
    paddingLeft: theme.spacing(4),
  },
}));

export default function SideNavigation() {
  const classes = useStyles();
  const [open, setOpen] = React.useState(true);

  const handleClick = () => {
    setOpen(!open);
  };

  return (

    <List
      component="nav"
      aria-labelledby="nested-list-subheader"
      className={classes.root}
    >
      <Divider variant="fullWidth" component="li"/>
      <ListItem button>
        <ListItemIcon>
          <ExploreIcon color="primary"/>
        </ListItemIcon>
        <ListItemText primary="Explore" color='inherit'/>
      </ListItem>
      <ListItem button>
        <ListItemIcon>
          <InboxIcon color="primary"/>
        </ListItemIcon>
        <ListItemText primary="Messages" />
      </ListItem>
      <ListItem button>
        <ListItemIcon>
          <NotificationsIcon color="primary"/>
        </ListItemIcon>
        <ListItemText primary="Notifications" />
      </ListItem>
      <ListItem button onClick={handleClick}>
        <ListItemIcon>
          <AccountBoxIcon color="primary"/>
        </ListItemIcon>
        <ListItemText primary="Profile" />
      </ListItem>
    </List>
  );
}