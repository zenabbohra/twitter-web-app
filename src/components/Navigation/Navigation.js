import React from 'react';
import SearchBox from './SearchBox';
import HomeIcon from '@material-ui/icons/Home';
import {Box} from "@material-ui/core";
import Link from "@material-ui/core/Link";
import Typography from "@material-ui/core/Typography";
import styles from './Navigation.module.css';

const Navigation = ({route, userName, onRouteChange }) => {
  const preventDefault = event => event.preventDefault();
    return <Box display="flex" alignItems="center" justifyContent="space-between" minHeight={48} marginLeft={2}>
      {
        route === 'home' ?
          <Typography variant="body1" color="primary" component="p">
            Welcome
            <Link href="#" onClick={ (event) => {preventDefault(event); onRouteChange('user profile')} } variant="h5" >
              {' ' + userName}
            </Link>
          </Typography> :
          <Link onClick={() => onRouteChange('home')} href="/" className={styles.link} color="primary" variant="body1">
            <HomeIcon className={styles.icon}/>
            Home
          </Link>
      }
      <Box display="flex" alignItems="center">
        <SearchBox/>
        <Box margin={2}>
          <Link href="#" onClick={ (event) => {preventDefault(event); onRouteChange('sign in')} } variant="h5">
            Sign out
          </Link>
        </Box>
      </Box>
    </Box>
};

export default Navigation;