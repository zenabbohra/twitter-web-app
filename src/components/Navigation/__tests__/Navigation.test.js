import React from 'react';
import { shallow } from 'enzyme';
import Navigation from '../Navigation';
import SearchBox from "../SearchBox";

describe('<Navigation/>', () => {
  describe('renders', () => {
    const subject = shallow(<Navigation/>);

    it('has one <HomeIcon/>', () => {
      const homeIcon =  subject.find('HomeIcon');
      expect(homeIcon.length).toBe(1);
      expect(homeIcon).toMatchSnapshot();
    });

    it('has one <SearchBox/>', ()=> {
      const searchBox = subject.find('SearchBox');
      expect(searchBox.length).toBe(1);
      expect(searchBox).toMatchSnapshot();
    });

    it('has three <Link/>', () => {
      const link = subject.find('WithStyles(ForwardRef(Link))');
      expect(link.length).toBe(2);
      expect(link).toMatchSnapshot();
    });

  })
});