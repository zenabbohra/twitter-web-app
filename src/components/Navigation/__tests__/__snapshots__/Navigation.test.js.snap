// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation/> renders has one <HomeIcon/> 1`] = `
<HomeIcon
  className="icon"
/>
`;

exports[`<Navigation/> renders has one <SearchBox/> 1`] = `<SearchBox />`;

exports[`<Navigation/> renders has three <Link/> 1`] = `
Array [
  <WithStyles(ForwardRef(Link))
    className="link"
    color="primary"
    href="/"
    onClick={[Function]}
    variant="body1"
  >
    <HomeIcon
      className="icon"
    />
    Home
  </WithStyles(ForwardRef(Link))>,
  <WithStyles(ForwardRef(Link))
    href="#"
    onClick={[Function]}
    variant="h5"
  >
    Sign out
  </WithStyles(ForwardRef(Link))>,
]
`;
